workshop/content/20-typescript/30-hello-cdk/300-cdk-watch.md

```diff
@@ -0,0 +1,179 @@
+{{% /notice %}}
+
+We can speed up that deployment time with `cdk watch`, which watches your
+CDK application and looks for hotswappable changes. Which `cdk watch` finds a
+change that can be hotswapped, it bypasses the CloudFormation deployment entirely
+and instead makes the necessary SDK call to update your code for you. Changes to 
+your AWS Lambda asset code, AWS Step Function State Machine definition, and other
+non-resource changes have the potential to be hotswapped. Here, we will try
+`cdk watch` with your AWS Lambda asset code.
+
+## Modify your `cdk.json` file
+
+When the `cdk watch` command runs, the files that it observes is determined by the
+`"watch"` setting in the `cdk.json` file. It has two sub-keys, `"include"` and
+`"exclude"`, each of which can be either a single string or an array of strings.
+Each entry is interpreted as a path relative to the location of the `cdk.json`
+file. Globs, both `*` and `**`, are allowed to be used.
+
+Your `cdk.json` file should look like this:
+
+```json
+{
+  "app": "npx ts-node --prefer-ts-exts bin/workshop.ts",
+  "watch": {
+    "include": [
+      "**"
+    ],
+    "exclude": [
+      "README.md",
+      "cdk*.json",
+      "**/*.d.ts",
+      "**/*.js",
+      "tsconfig.json",
+      "package*.json",
+      "yarn.lock",
+      "node_modules",
+      "test"
+    ]
+  },
+  "context": {
+    ...
+  }
+}
+```
+
+As you can see, the sample app comes with a suggested `"watch"` setting. We do in
+fact want to observe our `.js` files in the `lambda` folder, so let's remove
+`"**/*.js"` from the `"exclude"` list:
+
+```json
+{
+  "app": "npx ts-node --prefer-ts-exts bin/workshop.ts",
+  "watch": {
+    "include": [
+      "**"
+    ],
+    "exclude": [
+      "README.md",
+      "cdk*.json",
+      "**/*.d.ts",
+      "tsconfig.json",
+      "package*.json",
+      "yarn.lock",
+      "node_modules",
+      "test"
+    ]
+  },
+  "context": {
+    ...
+  }
+}
+```
+
+Now you're all set to start watching!
+
+## See how long 'CDK Watch' takes
+
+First, call `cdk watch`: 
+
+```bash
+cdk watch
+```
+
+This will trigger an initial deployment and immediately begin observing the files
+you've specified in `cdk.json`.
+
+For the purposes of this exercise, make sure that the initial deployment is
+finished before proceeding. Otherwise, the changes you make will be queued for
+deployment and released _after_ the first deployment finishes.
+
+Let's change our lambda asset code in `lambda/hello.js` one more time. It doesn't
+matter what you change it to, but let's say we want it to say
+`"Good Afternoon, CDK"` now.
+
+Once you click save or hit `CTRL+s`, `cdk watch` will recognize that your file has
+changed and trigger a new deployment. In this case, it will recognize that we can
+hotswap the lambda asset code, so it will bypass a CloudFormation deployment and
+deploy directly to the Lambda service instead.
+
+How fast did deployment take?
+
+```
+INSERT PHOTO HERE
+```
```

Comment:
I would modify this a bit, something like:

"Once you click save or hit `CTRL+s`, `cdk watch` will recognize that your file has changed and trigger a new deployment. 

```
INSERT PHOTO HERE
```

The warning message indicates that this lambda code change is hotswappable, so it will bypass CloudFormation and go directly to the Lambda Service API. This introduces drift between your CloudFormation template and your deployed application code. Because of this drift, hotswapping should never be used in a production environment. Hotswap deployments are faster, but lack the robust safety features of CloudFormation deployments. To disable hotswapping, pass the `--no-hotswap` flag to `cdk watch`."

Reply:
I feel like we addressed this already above, with hotswapping. I don't think it makes sense to repeat all of that here.

It's a good idea to mention `cdk watch --no-hotswap` though.